src/main/java/org/folio/dcb/service/impl/ServicePointServiceImpl.java

```diff
@@ -0,0 +1,58 @@
+package org.folio.dcb.service.impl;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.folio.dcb.client.feign.InventoryServicePointClient;
+import org.folio.dcb.domain.dto.DcbPickup;
+import org.folio.dcb.domain.dto.HoldShelfExpiryPeriod;
+import org.folio.dcb.domain.dto.ServicePointRequest;
+import org.folio.dcb.exception.ResourceAlreadyExistException;
+import org.folio.dcb.service.ServicePointService;
+import org.springframework.stereotype.Service;
+
+@Service
+@Log4j2
+@RequiredArgsConstructor
+public class ServicePointServiceImpl implements ServicePointService {
+
+  private final InventoryServicePointClient servicePointClient;
+
+  @Override
+  public ServicePointRequest createServicePoint(DcbPickup pickupServicePoint) {
+    log.debug("createServicePoint:: automate service point creation {} ", pickupServicePoint);
+
+    ServicePointRequest servicePointRequest = createServicePointRequest(
+      pickupServicePoint.getServicePointId(),
+      getServicePointName(pickupServicePoint.getLibraryName(), pickupServicePoint.getServicePointName()),
+      getServicePointCode(pickupServicePoint.getLibraryName(), pickupServicePoint.getServicePointName()),
+      getServicePointName(pickupServicePoint.getLibraryName(), pickupServicePoint.getServicePointName())
+    );
+
+     try{
+       return servicePointClient.createServicePoint(servicePointRequest);
+     } catch (ResourceAlreadyExistException e){
```

Comment:
I can see that you are converting the unprocessable entity into ResourceAlreadyexist exception in DcbErrorDecoder class. I think it will be good if we catch UNPROCESSABLE_ENTITY Exception here directly. So that we don't need lot of classes.  
@MagzhanArtykov  Please let me know your thoughts